tensorflow/core/grappler/optimizers/remapper.cc

```diff
@@ -1744,8 +1753,12 @@ bool RequiresInferredShapes(const RemapperContext& ctx, int node_index) {
 
 #ifdef INTEL_MKL
   (void)is_relu_biasadd_conv2d_candidate;  // To fix unused variable error.
-  return is_batch_norm_candidate() || is_batch_norm_fusion_candidate() ||
-         IsContractionWithAdd(ctx, node_index);
+  if (IsMKLEnabled())
+    return is_batch_norm_candidate() || is_batch_norm_fusion_candidate() ||
+           IsContractionWithAdd(ctx, node_index);
+  else
+    return is_relu_biasadd_conv2d_candidate() || is_batch_norm_candidate() ||
+           is_batch_norm_fusion_candidate();
 #else
   return is_relu_biasadd_conv2d_candidate() || is_batch_norm_candidate() ||
          is_batch_norm_fusion_candidate();
```

Comment:
Do we still need this `#ifdef INTEL_MKL`? Can the whole block just be:
```c++
  if (IsMKLEnabled()) {
    return is_batch_norm_candidate() || is_batch_norm_fusion_candidate() ||
           IsContractionWithAdd(ctx, node_index);
  }
  return is_relu_biasadd_conv2d_candidate() || is_batch_norm_candidate() ||
         is_batch_norm_fusion_candidate();
```
?
(We don't need `(void)is_relu_biasadd_conv2d_candidate;` anymore because the variable is now used.)

If you prefer to keep `#ifdef INTEL_MKL` and clean up later, then I think this should be fine:
```c++
#ifdef INTEL_MKL
  (void)is_relu_biasadd_conv2d_candidate;  // To fix unused variable error.
  if (IsMKLEnabled()) {
    return is_batch_norm_candidate() || is_batch_norm_fusion_candidate() ||
           IsContractionWithAdd(ctx, node_index);
  }
#endif  // INTEL_MKL
  return is_relu_biasadd_conv2d_candidate() || is_batch_norm_candidate() ||
         is_batch_norm_fusion_candidate();
```